### DISCUSSION TITLE
Best way to affecting the associated match weights for different comparisons?

### DISCUSSION THREAD

**Discussion by mmagoffin-sd** [moj-analytical-services/splink#767] [Q&A] [answered]
I've been successful in outputting linkage results for matching records, but I was wondering what the best way to affect how much a column is weighted would be? For example, looking through my possible matches from linker.predict(), I see that it looks like last names not matching is not pulling the overall match weight score down enough to make a match as unlikely as I would reasonably expect. Would a solution to this lie in manipulating the blocking rules or comparison levels somehow?

**Comment by RobinL** [comment, marked as answer]
You can manually set the m and u probabilities in your settings object if you want to override the estimated values.

For example you can see in the last cell [here](https://github.com/moj-analytical-services/splink_demos/blob/master/03_Estimating_model_parameters.ipynb) we're saving settings to json.

And then loading in the first cell [here](https://github.com/moj-analytical-services/splink_demos/blob/master/04_Predicting_results.ipynb)

And in the [saved settings](https://github.com/moj-analytical-services/splink_demos/blob/96b19b5a8285fd4254a7d2ff9c5eea7752c5ebbd/demo_settings/saved_model_from_demo.json#L16) you can see the m and u probabilities.  These are the values you'd want to change for a manual override.

More information [here](https://www.robinlinacre.com/understanding_match_weights/) on what the m and u probabilities mean

You can also manually set the m and u probabilities with code like:

```
# Get first comparison
comp = linker._settings_obj.comparisons[0]

# get first comparison level
cl = comp.comparison_levels[1]

# set m probability
cl.m_probability = 0.1
```
